yesod-test/Yesod/Test.hs

```diff
@@ -958,6 +959,40 @@ genericNameFromHTML match label html =
         name:_ -> Right name
     _ -> Left $ "More than one label contained " <> label
 
+
+-- | Choose a given value from a radio input, identified by the value of it's corresponding label.
+-- This is a very simple and naive function, which could fail in more complex scenarios. Potentially still quite useful.
+--
+-- Given this HTML, and we want to submit @color=Red@ to the server:
+--
+-- > <form method="POST">
+-- >   <label for="color-red">Red</label>
+-- >   <input id="color-red" name="color" value="Red" />
```

Comment:
I think typical radio input markup should look like this:

```html
<input id="color-red" name="color" value="red">
<label for="color-red">Red</label>
```

_i.e._, the input should come before the label. Also, self-closing tags don't need a trailing slash.